usage/launcher/src/main/java/brooklyn/launcher/BrooklynLauncher.java

```diff
@@ -682,6 +676,8 @@ private void initManagementContext() {
             brooklynProperties.addFromMap(brooklynAdditionalProperties);
         }
         
+        ((ManagementContextInternal)managementContext).setCatalogInitialization(catalogInitialization);
```

Comment:
Overall I'm ok with this, but I do have a gut feeling that I don't like us injecting the `catalogInitialization`.

It feels like we want to inject config to control the catalog initialization. That could be used to configure the actual catalogInitialization object, which would be purely internal.

Injecting a full-blown `CatalogInitialization` object is obviously more flexible, but the contract of what that object must do is not very well defined for it to be on an interface. I'd expect we'd only want to inject a different impl in tests, or in really extreme cases. If we really want to support that kind of thing, we should think seriously about a dependency injection framework.